layer2/oscal_generator.go

```diff
@@ -53,43 +53,56 @@ func (c *Catalog) ToOSCAL(controlHREF string) (oscal.Catalog, error) {
 
 		controls := []oscal.Control{}
 		for _, control := range family.Controls {
-			parts := []oscal.Part{}
-			for _, ar := range control.AssessmentRequirements {
-				parts = append(parts, oscal.Part{
-					Class: control.Id,
-					ID:    ar.Id,
-					Name:  ar.Id,
-					Ns:    "",
-					Parts: &[]oscal.Part{
-						{
-							ID:    ar.Id + ".R",
-							Name:  "recommendation",
-							Ns:    oscalUtils.GemaraNamespace,
-							Prose: ar.Recommendation,
-							Links: &[]oscal.Link{
-								{
-									Href: fmt.Sprintf(controlHREF, c.Metadata.Version, ar.Id),
-									Rel:  "canonical",
-								},
-							},
-						},
-					},
-					Prose: ar.Text,
-					Title: "",
-				})
-			}
-
 			newCtl := oscal.Control{
-				Class: family.Title,
+				Class: family.Id,
 				ID:    control.Id,
+				Title: strings.TrimSpace(control.Title),
 				Links: &[]oscal.Link{
 					{
 						Href: fmt.Sprintf(controlHREF, c.Metadata.Version, strings.ToLower(control.Id)),
 						Rel:  "canonical",
 					},
 				},
-				Parts: &parts,
-				Title: strings.TrimSpace(control.Title),
+			}
+
+			var subControls []oscal.Control
+			for _, ar := range control.AssessmentRequirements {
+				subControl := oscal.Control{
+					ID:    fmt.Sprintf("%s.%s", control.Id, ar.Id),
```

Comment:
I'm thinking the `ar.Id`  should just be used here and ID locations. By convention, the Assessment Requirement IDs are already expressed in this format. The outcome sub-control ID for this change is, for example, `OSPS-AC-02.OSPS-AC-02.01`
```suggestion
					ID:    ar.Id,
```

Reply:
Gotcha. Changes Implemented.